src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -138,6 +138,9 @@ public static class ODataBatchReaderExtensions
                 request.Body = bufferedStream;
             }
 
+            // remove the content length and content type headers in case the batchRequest item doesn't have a body
+            request.ContentLength = null;
+            request.ContentType = null;
```

Comment:
is there any risk?

Reply:
@xuzhg Initially, the headers from the batch request were being copied to the individual requests inside the batch. This led to some of the requests getting headers that didn't make sense, for example some GET requests inside the batch would end up having a Content-Length and Content-Type even if they did not have any content/body. But if the individual requests (e.g. POST and PUT) have a body, then they would overwrite the content headers with their own. So I don't believe there is a risk.

Reply:
I buy that applying Content-Length across all of the requests is unhelpful, and that copying Content-Type to GET/DELETE requests is similarly not helpful. I do think there may be some use in copying Content-Type from the overall batch request to individual requests within the batch that have content (PUT/POST/PATCH) but don't have Content-Type header specified, and *in those cases* not setting Content-Type could be a breaking change.

Reply:
I created a separate issue to discuss/address this: https://github.com/OData/WebApi/issues/2055